include/envoy/ssl/private_key/private_key.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+#include <functional>
+#include <string>
+
+#include "envoy/api/v2/auth/cert.pb.h"
+#include "envoy/common/pure.h"
+#include "envoy/event/dispatcher.h"
+#include "envoy/ssl/private_key/private_key_callbacks.h"
+
+#include "openssl/ssl.h"
+
+namespace Envoy {
+namespace Server {
+namespace Configuration {
+class TransportSocketFactoryContext;
```

Comment:
Is this to break an include dependency loop?

Reply:
Yes, that's why this is needed -- Bazel will complain if this is done with a real dependency.

Reply:
Arguably we could do some header refactoring to fix this, but I think this is OK if this is a one off.

Reply:
Ok. Just for the record, the dep loop would be this:

        //include/envoy/ssl:context_manager_interface
        //include/envoy/ssl:context_config_interface
    .-> //include/envoy/ssl:tls_certificate_config_interface
    |   //include/envoy/ssl/private_key:private_key_interface
    |   //include/envoy/server:transport_socket_config_interface
    |   //include/envoy/secret:secret_manager_interface
    |   //include/envoy/secret:secret_provider_interface
    `-- //include/envoy/ssl:tls_certificate_config_interface